user/apis/views.py

```diff
@@ -1,10 +1,22 @@
 from rest_framework.decorators import api_view, permission_classes
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
+from rest_framework import generics
+from rest_framework_simplejwt.authentication import JWTAuthentication
+
+from user.models import User
+from user.apis.serializers import UserSerializer
+from user.apis.permissions import CustomPermissionMixin
+
 
 @api_view(['GET'])
 @permission_classes([IsAuthenticated])
 def protected_view(request):
     user = request.user
-    # Your logic for the protected view
     return Response({'message': f'Hello, {user.username}! This is a protected view.'})
+
+
+class UserListCreateView(CustomPermissionMixin, generics.ListCreateAPIView):
```

Comment:
use retrieve api view, and a owner of the account should be able to retrieve there details and no one else should be allowed